src/resources/containers.jl

```diff
@@ -1,5 +1,5 @@
 struct ContainerKey{N<:Real} <: ResourceKey
-  priority :: Int
+  priority :: Number
```

Comment:
This struct now has abstract typed slots, which would lead to boxing/allocations/inference issues. This would be better if we add a new type parameter (similar to how currently there is an `N` type parameter).

```julia
julia> struct A
           i::Number
       end

julia> isbits(A(1))
false

julia> isbitstype(typeof(A(1)))
false

julia> struct B{T<:Number}
           i::T
       end

julia> isbits(B(1))
true

julia> isbitstype(typeof(B(1)))
true

```

Reply:
ok. I'll update this on all the structs

Reply:
@Krastanov we can have the priorities be `Real` instead of `Number` and use the type parameter `N` that is already being used for the `amount` field. In reality, priorities can't be imaginary numbers, so `Real` works as well. What are your thoughts?

Reply:
Not sure. Is there a situation in which it is important for the amount to be integer and for the priority to be float? If we have the same type parameter, that forces both of these properties to be the same type. I think it is safer to use two separate parameters.

Reply:
ah yes. good call